compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -163,27 +153,30 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 //
                 // [block: If(lhs)] -true-> [else_block: dest = (rhs)]
                 //        | (false)
-                //  [shortcurcuit_block: dest = false]
+                //  [shortcircuit_block: dest = false]
                 //
                 // Or:
                 //
                 // [block: If(lhs)] -false-> [else_block: dest = (rhs)]
                 //        | (true)
-                //  [shortcurcuit_block: dest = true]
+                // [shortcircuit_block: dest = true]
 
-                let (shortcircuit_block, mut else_block, join_block) = (
-                    this.cfg.start_new_block(),
-                    this.cfg.start_new_block(),
-                    this.cfg.start_new_block(),
-                );
+                let local_scope = this.local_scope();
```

Comment:
The changes from here to the end of the file can be reverted to address the lang team's comment. r=me with that done.

Reply:
Done